### ISSUE TITLE
indent multiline logical expression seams to be default to '0'

### ISSUE THREAD

**Issue by gombek** [eslint/eslint#10253]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 8.7.0
* **npm Version:** 5.6.0
* **yarn Version:** 1.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "extends": ["airbnb-base", "prettier", "plugin:jest/recommended"],
  "env": {
    "browser": true,
    "node": true,
    "es6": true,
    "jest/globals": true,
  },
  "plugins": ["jest", "prettier"]
  "rules": {
    "prettier/prettier": "error",
    "complexity": ["error", 6],
    "function-paren-newline": "off",
    "new-cap": "error",
    "no-use-before-define": "off",
    "no-restricted-syntax": ["error", "ForInStatement", "LabeledStatement", "WithStatement"],
    "object-curly-newline": "off",
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": [".config/**/*", "**/*.{test,spec}.js"],
        "peerDependencies": true
      }
    ],
    "import/prefer-default-export": "warn"
    "indent": [
      "error",
      2,
      {
        "SwitchCase": 1
      }
    ]
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
        {arrows({ device, touchable }) &&
          overflow !== 'none' && [
            <CarouselArrow
              key="arrow-left"
              direction="left"
              onClick={this._onLeftArrowClick}
              active={overflow === 'left' || overflow === 'both'}
              theme={theme}
            />,
            <CarouselArrow
              key="arrow-right"
              direction="right"
              onClick={this._onRightArrowClick}
              active={overflow === 'right' || overflow === 'both'}
              theme={theme}
            />,
          ]}
```

```bash
eslint \".config\" \"packages/**/@(src|test|stories)/**/*.@(js|jsx)\"
```

**What did you expect to happen?**
no error.

**What actually happened? Please include the actual, raw output from ESLint.**
  167:1  error  Expected indentation of 10 spaces but found 12  indent
  168:1  error  Expected indentation of 12 spaces but found 14  indent
  169:1  error  Expected indentation of 12 spaces but found 14  indent
  170:1  error  Expected indentation of 12 spaces but found 14  indent
  171:1  error  Expected indentation of 12 spaces but found 14  indent
  172:1  error  Expected indentation of 12 spaces but found 14  indent
  173:1  error  Expected indentation of 10 spaces but found 12  indent
  174:1  error  Expected indentation of 10 spaces but found 12  indent
  175:1  error  Expected indentation of 12 spaces but found 14  indent
  176:1  error  Expected indentation of 12 spaces but found 14  indent
  177:1  error  Expected indentation of 12 spaces but found 14  indent
  178:1  error  Expected indentation of 12 spaces but found 14  indent
  179:1  error  Expected indentation of 12 spaces but found 14  indent
  180:1  error  Expected indentation of 10 spaces but found 12  indent
  181:1  error  Expected indentation of 8 spaces but found 10   indent
